notebook/static/notebook/js/notebook.js

```diff
@@ -1958,26 +1958,48 @@ define(function (require) {
         dialog.modal(options.dialog);
         return promise;
     };
-    
+
     /**
-     * Execute or render cell outputs and go into command mode.
+     * Execute cells corresponding to the given indices.
+     *
+     * @param {list} indices - indices of the cells to execute
      */
-    Notebook.prototype.execute_cell = function () {
-        // mode = shift, ctrl, alt
-        var cell = this.get_selected_cell();
-        
-        cell.execute();
+    Notebook.prototype.execute_cells = function (indices) {
+        if (indices.length === 0) {
+            return;
+        }
+
         this.command_mode();
         this.set_dirty(true);
+
+        var cell;
+        for (var i = 0; i < indices.length; i++) {
+            this.select(indices[i]);
+            cell = this.get_cell(indices[i]);
+            cell.execute();
+        }
+    };
+
+    /**
+     * Execute or render cell outputs and go into command mode.
+     */
+    Notebook.prototype.execute_cell = function () {
+        this.execute_cells(this.get_marked_indices());
     };
 
     /**
      * Execute or render cell outputs and insert a new cell below.
      */
     Notebook.prototype.execute_cell_and_insert_below = function () {
+        // execute the marked cells, and don't insert anything
+        var indices = this.get_marked_indices();
```

Comment:
We need the logic here of checking how many marked cells there are -- if there are multiple, then we want to execute the marked cells, otherwise we want the normal behavior. I could have it call `execute_marked_cells` (aka `execute_cell`) as well as doing the check but then it has to call `get_marked_indices` twice.